### PR TITLE
[Tests] Fix scenario where nethermind stops before capturing extra logs - convert script to Python

### PR DESCRIPTION
**Pull request by kamilchodola**
Fix scenario where nethermind stops before capturing extra logs

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [X] Other: _Tests_

#### Requires testing

- [ ] Yes
- [X] No

#### If yes, did you write tests?

- [ ] Yes
- [X] No

### CODE REVIEW

**Review by LukaszRozmej — APPROVED** [review]

File: scripts/waitForSync.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            counter += 1
+            if counter >= 100:
+                print("Exiting after capturing extra logs due to error.")
+                sys.exit(1)
+            continue
+
+        if any(bad_log in line for bad_log in bad_logs):
+            print(f"Error: Found bad log in line: {line.strip()}")
+            found_bad_log = True
+            continue
```
Comment (brbrr) [inline scripts/waitForSync.py:45]: Not sure I understand why it's implemented this way. From my understanding:

When a bad log is found, we'll set the flag to true;
After that, we'll run the for loop 100 more times, and then exit.

Is it used to capture extra logs?

File: scripts/waitForSync.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import os
```
Comment (brbrr) [inline scripts/waitForSync.py:1]: Not sure, but was wondering if we could reuse the script from post-merge tests, what do you think? Probably not needed at this point, I guess?

**Review by brbrr — APPROVED** [review]
Looks good overal, but let few minor comments for you to decide.

File: scripts/waitForSync.py (hunk 1 above)
Comment (kamilchodola) [reply scripts/waitForSync.py:45]: Yes exactly for this purpose

File: scripts/waitForSync.py (hunk 2 above)
Comment (kamilchodola) [reply scripts/waitForSync.py:1]: I reached the limit of optimizing to various ise cases SH script and python gave me more flexibility and went quickly with rewrite.

I have no issue reusing existing one and keep them identical to some extent!

**Pull request merged**